curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md

```diff
@@ -17,439 +17,445 @@ Answer all of the questions below correctly to pass the quiz.
+To improve performance
 
 ---
 
-Placeholder distractor 3
+To optimize rendering time
 
 #### --answer--
 
-Placeholder answer
+To reduce code complexity
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+In which scenario might defining a custom property in a media query cause unexpected behavior?
 
 #### --distractors--
 
-Placeholder distractor 1
+When the property is defined using a `px` unit
 
 ---
 
-Placeholder distractor 2
+When the property is defined outside of the `:root`
 
 ---
 
-Placeholder distractor 3
+When the property is defined at the bottom of the stylesheet
 
 #### --answer--
 
-Placeholder answer
+When the media query condition no longer applies, and the custom property is no longer valid
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What’s a key drawback when using CSS variables in combination with older layout models like `float` or `table`?
 
 #### --distractors--
 
-Placeholder distractor 1
+They cannot be referenced by the `float` property
 
 ---
 
-Placeholder distractor 2
+They may cause significant reflows, impacting performance
 
 ---
 
-Placeholder distractor 3
+They require JavaScript to function correctly
 
 #### --answer--
 
-Placeholder answer
+Older layout models may not properly inherit or apply custom properties, causing inconsistencies
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+When using custom properties with CSS Grid or Flexbox, what should you be cautious about?
 
 #### --distractors--
 
-Placeholder distractor 1
+Custom properties override grid or flex properties by default
 
 ---
 
-Placeholder distractor 2
+Flexbox and Grid ignore custom properties during layout calculations
 
 ---
 
-Placeholder distractor 3
+Custom properties can cause layout shifts when recalculated
 
 #### --answer--
 
-Placeholder answer
+Custom properties should be managed carefully, as they are recalculated with each layout pass, potentially impacting performance
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+If you define a custom property `--text-color` inside a media query and use `var(--text-color)` in multiple elements, what happens if the viewport changes and the media query no longer matches?
 
 #### --distractors--
 
-Placeholder distractor 1
+All the text that used the variable will revert to the default browser color
 
 ---
 
-Placeholder distractor 2
+The text will stay the same until the page is reloaded
 
 ---
 
-Placeholder distractor 3
+The custom property will persist even if the media query conditions no longer apply
 
 #### --answer--
 
-Placeholder answer
+The variable will no longer be available, and text will fall back to inherited or default styles
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following is the correct way to define a fallback value for a CSS variable when the custom property is not available?
 
 #### --distractors--
 
-Placeholder distractor 1
+`color: var(--primary-color, inherit);`
```

Comment:
This is the same as correct answer, distractors must be different.

Reply:
resolved in new commit